### ISSUE TITLE
help request: proxy rewrite rule does not work

### ISSUE THREAD

**Issue by brunopadz** [apache/apisix#11344]
### Description

Hi all!
I've been trying unsuccessfully for almost a week to manage proxy-rewrite to work as expected.

TLDR: I wanna receive a request on `https://api.example.com/v1/myapi/resources` and send to my backend as `/internal/api/resources`. But APISix sends `/v1/myapi/resources` to the backend. It simply does not work.

Here's the route config:

```json
{
    "value": {
        "plugins": {
            "proxy-rewrite": {
                "uri": "/internal/api/resources"
            }
        },
        "uri": "/v1/myapi/resources",
        "upstream_id": "516582780553921192",
        "methods": [
            "GET"
        ],
        "create_time": 1717787055,
        "id": "517170190664860322",
        "name": "api-test",
        "update_time": 1717961912,
        "status": 1
    },
    "key": "/apisix/routes/517170190664860322",
    "createdIndex": 316,
    "modifiedIndex": 324
}
```

More info:
- Running on EKS / Kubernetes 1.28
- Tried with ALB Ingress Controller exposing APISix endpoint. Also tried with Load Balancer (NLB with TLS Termination) as Service Type and no luck.

### Environment

- APISIX version (run `apisix version`): 3.9.0
- Operating system (run `uname -a`): Kubernetes / EKS
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): 1.25.3.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):  3.5.0
- APISIX Dashboard version, if relevant: 3.9.0
- APISix Helm Chart version: 2.8.0

**Comment by nfrankel** [comment]
You should probably use the `regex_uri` configuration parameter.

Here's a snippet to remove the v1 prefix and leave the rest untouched:

```json
{
  "plugins": {
    "proxy-rewrite": {
      "regex_uri": ["/v1(.*)", "$1"]
    }
  }
}
```

**Comment by brunopadz** [comment]
Quick update here. It's working now.

I've tried to create a new route using the API and it returned `unknown plugin [proxy-rewrite]`, so I added the `proxy-rewrite` to the plugin list on my values file and it loaded properly.

It seems the dashboard and the APISIX admin API could use some improvements. The dashboard allows for creating rewrite rules, but the API does not, as the plugin isn't listed in the available plugins.
